cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternalsCommon.qll

```diff
@@ -501,6 +505,77 @@ private module Cached {
     )
   }
 
+  private predicate isSource(Instruction instr, Operand iteratorAddress, int numberOfLoads) {
+    getAUse(instr) = iteratorAddress and
+    exists(BaseSourceVariableInstruction iteratorBase |
+      iteratorBase.getResultType() instanceof Interfaces::Iterator and
+      not iteratorBase.getResultType() instanceof Cpp::PointerType and
+      isUse(_, iteratorAddress, iteratorBase, numberOfLoads - 1, 0)
+    )
+  }
+
+  private predicate isSink(Instruction instr, CallInstruction call) {
+    getAUse(instr).(ArgumentOperand).getCall() = call and
+    // Don't include various operations that don't modify what the iterator points to.
```

Comment:
I assume you mean: these calls might change the iterator itself, but they will never change the object that the iterator points to?

The sentence is currently a bit ambiguous: is it talking about the iterator being modified, or the object that the iterator points to being modified?

Reply:
> I assume you mean: these calls might change the iterator itself, but they will never change the object that the iterator points to?

Yes, that is correct. I'll rephrase the sentence to be less ambiguous.

Reply:
Hopefully 0a6f914bfcbef3b147fa5817d33d24f995ac17cf makes this more clear 🤞.